stl/inc/xutility

```diff
@@ -3246,10 +3249,38 @@ namespace ranges {
         }
     };
 
-    // VARIABLE ranges::distance
     inline constexpr _Distance_fn distance{_Not_quite_object::_Construct_tag{}};
 
-    // CLASS ranges::_Next_fn
+    // VARIABLE ranges::ssize
+    class _Ssize_fn { // Per the proposed resolution of LWG-3403
+    public:
+        // clang-format off
+        template <class _Rng>
+        _NODISCARD constexpr auto operator()(_Rng&& _Range) const requires requires { _RANGES size(_Range); } {
+            using _SSSsssigned = _Make_signed_like_t<decltype(_RANGES size(_Range))>;
+            using _Ty          = typename _Parsel_type<is_integral_v<_SSSsssigned>>::template _Apply<_SSSsssigned>;
+            return static_cast<_Ty>(_RANGES size(_Range));
+        }
+        // clang-format on
+    private:
+        template <bool _IsIntegral>
+        struct _Parsel_type {
+            template <class _Ty>
+            using _Apply = common_type_t<ptrdiff_t, _Ty>;

```

Comment:
Could we do something like `common_type_t<conditional_t<is_integral_v<S>, ptrdiff_t, S>, S>` inline above?